File: jmolecules-axon/02-axonstyle-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/api/command/CreateBankAccountCommand.java
package org.jmolecules.example.axonframework.domain.api.command;

import org.axonframework.modelling.command.TargetAggregateIdentifier;
import org.jmolecules.architecture.cqrs.annotation.Command;
import org.jmolecules.ddd.annotation.Association;

@Command(namespace = "axon.bank", name = "CreateBankAccountCommand")
public record CreateBankAccountCommand(
  @Association
  String accountId,
  int initialBalance
) {
  // empty
}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/domain/api/command/CreateBankAccountCommand.kt
package org.jmolecules.example.axonframework.domain.api.command

import org.jmolecules.architecture.cqrs.annotation.Command
import org.jmolecules.ddd.annotation.Association
import org.jmolecules.example.axonframework.domain.api.type.AccountId
import org.jmolecules.example.axonframework.domain.api.type.Balance

@Command(namespace = "axon.bank", name = "CreateBankAccountCommand")
data class CreateBankAccountCommand(
    @Association
    val accountId: AccountId,
    val initialBalance: Balance
)
